src/tree_data.c

```diff
@@ -2779,6 +2779,82 @@ lyd_target(const struct ly_path *path, const struct lyd_node *tree)
     return (struct lyd_node_term *)target;
 }
 
+/**
+ * @brief Compare schemas of two Compiled YANG data nodes.
+ *
+ * @param schema1 of first node.
+ * @param schema2 of second node.
+ * @return 0 if the schemas are equivalent otherwise 1.
+ */
+static ly_bool
+lyd_compare_schemas(const struct lysc_node *schema1, const struct lysc_node *schema2)
+{
+    if (schema1 == schema2) {
+        return 0;
+    } else if (!schema1 || !schema2) {
+        return 1;
+    }
+
+    if (schema1->nodetype != schema2->nodetype) {
+        return 1;
+    }
+
+    if (schema1->module->latest_revision != schema2->module->latest_revision) {
+        return 1;
+    }
+
+    if (strcmp(schema1->name, schema2->name)) {
+        return 1;
+    }
+
+    if (strcmp(schema1->module->name, schema2->module->name)) {
+        return 1;
+    }
+
+    if (schema1->module->revision || schema2->module->revision) {
+        if (!schema1->module->revision || !schema2->module->revision) {
+            return 1;
+        }
+        if (strcmp(schema1->module->revision, schema2->module->revision)) {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+/**
+ * @brief Compare all parental schemas of two data nodes.
+ *
+ * @param schema1 of first node.
+ * @param schema2 of second node.
+ * @return 0 if the all parental schemas are equivalent otherwise 1.
+ */
+static ly_bool
```

Comment:
You are returning a boolean so I would expect `1` to be equal, `0` to be different but it inverted. Also, the name `compare` would imply return values just like `strcmp` but they are not, so this prototype is confusing. My idea would be to keep the boolean with inverted values and rename the function to `lyd_compare_schema_parents_equal`. No need for `schemas` (you cannot compare a single schema), `lyd_compare` prefix means it is specific to `lyd_compare` functions (not sure it is, once it is used somewhere else, it should be renamed), and `equal` means it is a predicate (boolean return).

Reply:
ok